### ISSUE TITLE
shelly reversed state and reverse status

### ISSUE THREAD

**Issue by pirattte** [domoticz/domoticz#5414]
I have updated to the last version of domoticz (2022.2.14727), and I have this issue:

When the blind is closed:
- State: Open
- Percentage 100%
- Buttons are reversed

![imagen](https://user-images.githubusercontent.com/110498887/202001020-f10b7a34-17cf-4129-829a-c3b0296dd772.png)
![imagen](https://user-images.githubusercontent.com/110498887/202003048-ac480dbb-ed94-4192-b937-5c9cdddac05a.png)

I has restored my previus backup of domoticz (2022.1 (build 14453) and it works ok.

When the  blins is closed:

- State: Closed
- Percetage: 100%
- Buttons are ok.

![imagen](https://user-images.githubusercontent.com/110498887/202001772-a2d3d575-fd14-4edc-85a8-b3b11b30c6d4.png)

I remember that in the past, about july, i had the same issue. But in august you solved the issue. 

Please, can you look it?

thanks

**Comment by pirattte** [comment]
Does the same thing happen to someone with the same version of domoticz and the shelly plugin also updated to the latest version?

**Comment by didi31** [comment]
It seems to be the same issue than this one

https://github.com/domoticz/domoticz/issues/5395

**Comment by gizmocuz** [comment]
Did you also enabled the reverse option?
How to you connect the shelly to domoticz? Via MQTT Auto Discovery or via a plugin?
If this is done via a plugin, the plugin needs to be updated to reflect the current blind behaviour

https://domoticz.com/wiki/Blinds

**Comment by pirattte** [comment]
Hi, 

thank you for your help. 

I have the pluguin Shelly_MQTT, and I have the lastest version (from yesterday) of it.

![imagen](https://user-images.githubusercontent.com/110498887/202250932-51cf4e58-5a03-4638-80f9-68d3330c6501.png)


If i put the reversed option, I seems that all is ok, but not. Because when the blind is open, the percentage are closed.

Thanks

**Comment by svde** [comment]
Please try to apply this diff [https://github.com/enesbcs/Shelly_MQTT/commit/311551643e915e8e5817d108ce8095e5a6d3aee0#commitcomment-90148079](url) see if that also works for you.

**Comment by kitopopo** [comment]
> I have updated to the last version of domoticz (2022.2.14727), and I have this issue:
> 
> When the blind is closed:
> 
>     * State: Open
> 
>     * Percentage 100%
> 
>     * Buttons are reversed
> 
> 
> ![imagen](https://user-images.githubusercontent.com/110498887/202001020-f10b7a34-17cf-4129-829a-c3b0296dd772.png) ![imagen](https://user-images.githubusercontent.com/110498887/202003048-ac480dbb-ed94-4192-b937-5c9cdddac05a.png)
> 
> I has restored my previus backup of domoticz (2022.1 (build 14453) and it works ok.
> 
> When the blins is closed:
> 
>     * State: Closed
> 
>     * Percetage: 100%
> 
>     * Buttons are ok.
> 
> 
> ![imagen](https://user-images.githubusercontent.com/110498887/202001772-a2d3d575-fd14-4edc-85a8-b3b11b30c6d4.png)
> 
> I remember that in the past, about july, i had the same issue. But in august you solved the issue.
> 
> Please, can you look it?
> 
> thanks

Hi dear friends,

Same issue for me @gizmocuz @pirattte . I also have the plugin shelly_mqtt from enesbcs updated yesterday.
I wait news from you.

Thanks in advanced

Best regards.

**Comment by gizmocuz** [comment]
have you tried the patch from svde mentioned above

https://github.com/enesbcs/Shelly_MQTT/commit/311551643e915e8e5817d108ce8095e5a6d3aee0#commitcomment-90148079

**Comment by kitopopo** [comment]
> have you tried the patch from svde mentioned above
> 
> [enesbcs/Shelly_MQTT@3115516#commitcomment-90148079](https://github.com/enesbcs/Shelly_MQTT/commit/311551643e915e8e5817d108ce8095e5a6d3aee0#commitcomment-90148079)

Hi, thanks for your answer.

The  command "diff --git a/plugin.py b/plugin.py" doesn't work in a raspberri py.

Do you have any idea how I can apply the patch? because for my 5 shellys of the blinds to work I have to have them inverted, please see the attached pictures.

![blinds1](https://user-images.githubusercontent.com/51416982/203301138-7224d396-5842-4a79-82ab-973683f4b372.jpg)
![blinds2](https://user-images.githubusercontent.com/51416982/203301144-014ebd64-e879-4586-a10e-543cc6cbd8a3.jpg)

Apart from inverting the values ​​of the photos, I have to invert and modify all my variables and all my scripts (domoticz.devices('Persiana salon').switchOn() change by switchOff) , since open is closed and closed is open, and the percentage of the bar also changes from 100% open to 100% closed. It seems incredible to me to have to change everything when it has always worked correctly. Thank you I am waiting for your news

Thanks for your work. Best regards

**Comment by gizmocuz** [comment]
@kitopopo, that is correct, the blinds behavior has been reverted to match existing home automation platforms and products.
Until now people had to use inverted blinds, plugin developers needed to invert the logic inside their plugins for Domoticz etc etc
The new blinds behavior is explained on our wiki

https://www.domoticz.com/wiki/Blinds

For dzVents you need to use .open() and .close() and not switch on/off, as these commands are for lights and now for blinds

I know that this is a major change, but it needed to be done as blinds where always a pain

**Comment by svde** [comment]
Here you go, rename by removing .txt file extension. [plugin.py.txt](https://github.com/domoticz/domoticz/files/10067537/plugin.py.txt)

**Comment by waltervl** [comment]
Issue can be closed. Fixed in Shelly MQTT plugin: https://github.com/enesbcs/Shelly_MQTT/pull/96
Use the update plugin actions as mentioned in https://github.com/enesbcs/Shelly_MQTT

**Comment by svde** [comment]
> Issue can be closed. Fixed in Shelly MQTT plugin: [enesbcs/Shelly_MQTT#96](https://github.com/enesbcs/Shelly_MQTT/pull/96) Use the update plugin actions as mentioned in https://github.com/enesbcs/Shelly_MQTT

Did you test it?

On my system I need to invert it for it to be consistent, see patch mentioned in this [comment](https://github.com/enesbcs/Shelly_MQTT/commit/311551643e915e8e5817d108ce8095e5a6d3aee0#commitcomment-90148079).

**Comment by waltervl** [comment]
I cannot test but if it needs extra fixing it should be fixed and/or discussed in the plugin. You can prepare a PR in the plugin repository.

**Comment by kitopopo** [comment]
Dear @gizmocuz @pirattte 

Today after updating domoticz and the shelly plugin something has changed. 

Now I don't  need to tick the "reverse open/close state" box on all my devices with shellys. Now the order to "up blind" coincides with up. The order to "down blind" coincides with down. But **while the blind is going up, the **text** is wrong --> closed. And while the blind is lowering the **text** is also wrong --> open.** 
If I tick the "reverse position" box,  I don't get any change in the text. (Also Now the percentage bar is correct for me. All bar blue in the right means that the blind is opened to 100%.- Same in the shelly app .-)

But the **text problem** I don't know if the problem is from some domoticz update, from the shelly plugin or even from the zigbee2mqtt plugin (stas-demydiuk). Could you reopen this issue and check again how it works? I cannot be changing all my scripts and variables, there are too many and some confusion can cause serious problems in my awnings or blinds. 

Thanks in advance again. Receive a cordial greeting and thank you for your work and dedication

**Comment by waltervl** [comment]
Did you check wiki page https://domoticz.com/wiki/Blinds and did you switch the open/close state switch to get the TEXT correct?

**Comment by kitopopo** [comment]
Gracias por contestar @waltervl ,

Of course I have read and tried all the combinations and I can't find the right combination. 8 days ago when I wrote, yes, I could get a correct combination, but as I tell you, something has changed and it is not possible to get the correct text. 8 days ago I had to reverse all my shellys and not now, they all work in the right direction. someone has changed something in some update. The shelly plugin from enesbcs has been updated 3 days ago, Maybe worng update? changes? Thanks again. Best regards

**Comment by waltervl** [comment]
Something changed in the plugin.... Please discus there or in the forum. But it should be good for all.

**Comment by gizmocuz** [comment]
For sure nothing has changed in Domoticz related to blinds since the stable version. I guess you updated the shelly plugin

**Comment by kitopopo** [comment]
> Seguro que nada ha cambiado en Domoticz relacionado con las persianas desde la versión estable. Supongo que actualizaste el complemento Shelly.

Yes, I updated domoticz beta and shelly plugin at once.
I can see changes in the shelly plugin 4 days ago. I will open an issue towards the plugin to @enesbcs for you to review. Thanks again. Best regards

**Comment by kitopopo** [comment]
Dear @gizmocuz ,

Apart from the fact that the drawing of the action in blue color and that the text for raising and lowering the shelly devices is wrong and cannot be corrected by touching the options, I have found that in **the android domoticz app all the blinds are inverted with respect to the domoticz web server**. We have to hit "blind up" so that the blind goes down. It seems incredible to me that in an application in which almost all of us have bought we have this problem.

Apart from all this, and always from my point of view, it seems horrible to me that Mr. enesbcs no longer support the plugin . On the plugin page it says this: _"In 2022 I am using Domoticz MQTT AutoDiscovery hardware for my own Shelly devices, so further improvement of this plugin is now ended. I may publish a script which teaches shellies to AD in the future, depending on amount of my spare time.PR's can be added until i will archive this project, but will be untested"._

The real problem is that the autodiscovery of domoticz today does not work as it should and it needs to be polished and outlined so that it works correctly, meanwhile and in my opinion, ¿is it not possible that someone continues to support the plugin of the shelly until get the native autodiscovery works perfectly? Please at least support the shelly plugin if possible while you improve autodiscovery to be able to do the migration. Now all of us, those of us who have not yet made the leap to the new autodiscovery, are left hanging and without support. When an easy solution would be for both integrations to work in parallel until the improved version of autodiscovery

I know many people who, thanks to no good domoticz integrations, are migrating to the competition, at the moment I am still here with you and many other people, but please solve these small problems if possible until autodiscovery works well (I have read that to this day it still does not detect the new devices shelly plus and shelly plus 2pm. 

Thanks in advance, and all this writing is a personal opinion. You are the one in charge and make the decisions, I can only give my opinion. Despite these little problems I am still very happy with domoticz 

Thanks for your work, best regards

**Comment by gizmocuz** [comment]
First of all, is the Domoticz android app sending open/close or maybe has the inverted logic inside wrong?
Second, MQTT-AD is working quite well.
And thanks to your positive words we keep trying to make Domoticz an even better application
Happy holiday wishes!

**Comment by kitopopo** [comment]
> First of all, is the Domoticz android app sending open/close or maybe has the inverted logic inside wrong? Second, MQTT-AD is working quite well. And thanks to your positive words we keep trying to make Domoticz an even better application Happy holiday wishes!

Thanks for your answer, in domoticz server , my blinds works fine and correctly. When I press "up" the blind goes up and vice versa. I don't have the boxes for "invert open/close state" or "reverse estate" checked. After the last update of @enesbcs plugin, I found that the text is incorrect, "going up" is going down and "going down" is going up. The blue color of the drawing of the blind that lights up when the blind is in motion is also inverted. 
Having all this working more or less well in domoticz server (with the exception of the problems indicated) in the android app all the blinds are inverted and in the app I cannot select invert open/close state. This can only be selected on the server. But if I invert it on the server then the blinds work inverted on the server and in the app they work correctly.,

thanks in advenced

**Comment by gizmocuz** [comment]
If you mean by 'server' the web gui inside domoticz, then yes, it is working correctly.
If you use a external application to control domoticz, and this is inversed, please contact the authors of this application and ask them if they could have a look at it

**Comment by kitopopo** [comment]
> If you mean by 'server' the web gui inside domoticz, then yes, it is working correctly.
> If you use a external application to control domoticz, and this is inversed, please contact the authors of this application and ask them if they could have a look at it

Dear friend, thanks for your answer. I use domoticz web gui , the shellys are working fine but with a text problem using the external plugin from @enesbcs. Also fails the blue colour that switches on when i make click in down or up. This blue colour is inverted.
I use native app domoticz from android. ( i paid for this version).  In the native app domoticz the blinds are inverted. When i click in "up blind" the blind goes down. 
Do you want 2 videos in order to verify that my info is real? The blinds in the native app problem are inverted with respect to the domoticz web gui. 
I have been using domoticz for more of 3 or 4 years and i recognize the problems. Please , solve this problem with @enesbcs help. We would be very grateful. 


Thanks in advanced. Best regards

**Comment by gizmocuz** [comment]
@kitopopo , the blinds are working correctly inside Domoticz. They now follow the same logic as other home automation systems (like HASS)
Since the new stable release, plugins need to be updated and also the external apps
For example, there is no On/Off anymore. Only Open/Close.

The working of the blinds are explained here:
https://www.domoticz.com/wiki/Blinds

The logic we use now will stay (as it stayed for the last decade... but it was inverted)

I don't know if the Shelly devices follow the MQTT Auto Discovery protocol, if they do, you do not need a plugin.
If the don't, you might want to tweak the plugin, the code is not that complicated, most of the code in the plugin is to connect to MQTT and configuration logic

**Comment by kitopopo** [comment]
> @kitopopo , the blinds are working correctly inside Domoticz. They now follow the same logic as other home automation systems (like HASS) Since the new stable release, plugins need to be updated and also the external apps For example, there is no On/Off anymore. Only Open/Close.
> 
> The working of the blinds are explained here: https://www.domoticz.com/wiki/Blinds
> 
> The logic we use now will stay (as it stayed for the last decade... but it was inverted)
> 
> I don't know if the Shelly devices follow the MQTT Auto Discovery protocol, if they do, you do not need a plugin. If the don't, you might want to tweak the plugin, the code is not that complicated, most of the code in the plugin is to connect to MQTT and configuration logic

Dear friend,

I understand that they have changed the way shelly works, I am not discussing this. 
I have changed all my scripts and variables to match the new way of working of the shelly.
But @enesbcs made the last 2 updates to the shelly plugin, in the penultimate update everything worked fine and the reverse of how it worked a decade ago (it was necessary to reverse all in domoticz making click in the boxes "reverse open/close state") but in the last update of all something changed and it is no longer necessary to make the reverse in domoticz and all works in the correct direction in domoticz web gui. For this reason in the android domoticz app now works the other way around. 
there has been an erroneous update by them and they don't want to check ito

I take this opportunity to tell you that if they now work fine in domoticz web gui, in the domoticz android app they are reversed using the plugin shelly. (app for which we all paid for a long time). I am not going to discuss the same thing anymore, in the end there is no other option than to migrate to autodiscovery or i will y try to modify the shelly plugin personally. Thanks for your help. . Thanks for your time and dedication. a cordial greeting

**Comment by gizmocuz** [comment]
Please note that Domoticz is not the paid Domoticz android app so I cannot change that
If the domoticz android app is working incorrectly, please ask the developers of the application to correct this
And sure, if the shelly works natively with MQTT-AD, why not migrate right now! Add the MQTT-AD hardware in domoticz and just test it

**Comment by kitopopo** [comment]
> Tenga en cuenta que Domoticz no es la aplicación de Android de pago de Domoticz, por lo que no puedo cambiarla . Si la aplicación de Android de domoticz funciona incorrectamente, solicite a los desarrolladores de la aplicación que corrijan esto . Y seguro, si Shelly funciona de forma nativa con MQTT-AD, ¿por qué no? migrar ahora mismo! Agregue el hardware MQTT-AD en domoticz y simplemente pruébelo

Hi, the android paid domoticz app has always been the native domoticz android app. Has something changed? 3 years ago I bought it and it has always been the native application. I see that it is no longer in the play store, but we have always had the lite (free) version and the paid version.  My version is domoticz home automation and this application is the native domoticz for android. always has been the native android app.  My application was precisely the one that could include the "widgets" and you always supported this app

**Comment by gizmocuz** [comment]
Sure it is a very nice application, but what I meant is that it is not 'Domoticz'. So if an application or script needs to be adjusted, they have to do this
I myself am using the web interface of Domoticz also for all my mobile/handheld devices

**Comment by kitopopo** [comment]
> Sure it is a very nice application, but what I meant is that it is not 'Domoticz'. So if an application or script needs to be adjusted, they have to do this I myself am using the web interface of Domoticz also for all my mobile/handheld devices

If you have always supported this app, why now you no longer offer support? I do not understand anything

**Comment by gizmocuz** [comment]
I think you are mixing things up.
We have nothing to do with mobile applications, Domoticz is a standalone system and other write applications/plugins around it
So it is up to the maintainers of these applications/plugins to maintain it. (And this is the first time it has changed since decades)

**Comment by kitopopo** [comment]
Dear @gizmocuz ;

I have been testing older versions of the shelly plugin and exactly using the version 311551643e915e8e5817d108ce8095e5a6d3aee0 from November 15th the plugin works correctly, the text is correct in domoticz and the blue color of the action also is correct. 

Is necessary have to mark the boxes for reverse open/close state and reverse position so that everything works well and same as in the shelly app. The only inconvenience I have found is that in the domoticz android app the blinds work inverted (because we have marked the reverse Open/close state box)

In any case, I will try the autodiscovery due to that the shelly plugin will no longer receive support.

thanks and Best regards

**Comment by gizmocuz** [comment]
- As mentioned, the code in the python plugin is not large and even for someone who has never seen python before it should be not hard to modify it
- It could also be that the Android application needs to be modified
- if the shelly's work natively with MQTT-AD, yes go this path.
